src/Executables/Atlas/3. Configuration/5. Security/Core Isolation (VBS)/Current Configuration.ps1

```diff
@@ -18,26 +18,53 @@ if ($DisableAllVBS) {
 		New-ItemProperty -Path $memIntegrity -Name "Enabled" -Value 0 -PropertyType DWORD -Force
 		Remove-ItemProperty -Path $memIntegrity -Name "ChangedInBootCycle" -ErrorAction SilentlyContinue
 		Remove-ItemProperty -Path $memIntegrity -Name "WasEnabledBy" -ErrorAction SilentlyContinue
+		New-ItemProperty -Path $memIntegrity -Name "HVCIMATRequired" -Value 0 -PropertyType DWORD -Force
+		New-ItemProperty -Path $memIntegrity -Name "Locked" -Value 0 -PropertyType DWORD -Force
 	}
 
 	# Kernel-mode Hardware-enforced Stack Protection (Windows 11 only)
 	if (Test-Path $kernelShadowStacks) {
 		New-ItemProperty -Path $kernelShadowStacks -Name "Enabled" -Value 0 -PropertyType DWORD -Force
 		Remove-ItemProperty -Path $kernelShadowStacks -Name "ChangedInBootCycle" -ErrorAction SilentlyContinue
 		Remove-ItemProperty -Path $kernelShadowStacks -Name "WasEnabledBy" -ErrorAction SilentlyContinue
+		New-ItemProperty -Path $kernelShadowStacks -Name "HVCIMATRequired" -Value 0 -PropertyType DWORD -Force
+		New-ItemProperty -Path $kernelShadowStacks -Name "Locked" -Value 0 -PropertyType DWORD -Force
 	}
 
 	# Credential Guard (Windows 11 only)
 	if (Test-Path $credentialGuard) {
 		New-ItemProperty -Path $credentialGuard -Name "Enabled" -Value 0 -PropertyType DWORD -Force
 		Remove-ItemProperty -Path $credentialGuard -Name "ChangedInBootCycle" -ErrorAction SilentlyContinue
 		Remove-ItemProperty -Path $credentialGuard -Name "WasEnabledBy" -ErrorAction SilentlyContinue
+		New-ItemProperty -Path $credentialGuard -Name "HVCIMATRequired" -Value 0 -PropertyType DWORD -Force
+		New-ItemProperty -Path $credentialGuard -Name "Locked" -Value 0 -PropertyType DWORD -Force
 	}
 	exit
 } elseif ($EnableMemoryIntegrity) {
 	Write-Warning "Enabling memory integrity..."
-	Set-ItemProperty -Path $memIntegrity -Name "Enabled" -Value 1 -Type DWord
-	Set-ItemProperty -Path $memIntegrity -Name "WasEnabledBy" -Value 2 -Type DWord
+	if (Test-Path $memIntegrity) {
+		Set-ItemProperty -Path $memIntegrity -Name "Enabled" -Value 1 -Type DWord
+		Set-ItemProperty -Path $memIntegrity -Name "HVCIMATRequired" -Value 1 -Type DWord
+		Set-ItemProperty -Path $memIntegrity -Name "Locked" -Value 1 -Type DWord
+		Set-ItemProperty -Path $memIntegrity -Name "WasEnabledBy" -Value 2 -Type DWord
+		Set-ItemProperty -Path $memIntegrity -Name "ChangedInBootCycle" -Value 2 -Type DWord
+	}
+	# Kernel-mode Hardware-enforced Stack Protection (Windows 11 only)
+	if (Test-Path $kernelShadowStacks) {
```

Comment:
Windows doesn't create these keys (even if the feature is available) until you toggle it once, so if the user has never touched 'Kernel-mode Hardware-enforced Stack Protection' before, it won't be enabled even if it's available.

The `if` statement should therefore be removed.